src/bf_dpu_update.py

```diff
@@ -333,6 +333,54 @@ def get_push_uri(self):
         return (multi_part_uri, deprecated_uri)
 
 
+    def get_update_service_state(self):
+        url = self._get_url_base() + '/UpdateService'
+        response = self._http_get(url)
+        self.log('Get UpdateService state', response)
+        self._handle_status_code(response, [200])
+
+        state = ''
+        '''
+        {
+          ...
+          "Status": {
+            "Conditions": [],
+            "State": "Enabled"
+          }
+        }
+        '''
+        try:
+            state = response.json()['Status']['State']
+        except Exception as e:
+            raise Err_Exception(Err_Num.BAD_RESPONSE_FORMAT, 'Failed to extract update service state')
+        return state
+
+
+    def wait_update_service_ready(self):
+        if 'Enabled' == self.get_update_service_state():
+            return
+
+        print("Wait for update service ready")
+        timeout = 60 * 3 # Wait up to 3 minutes
```

Comment:
if enough if switch bios update firstly  as well?

Reply:
Tested, looks OK.

Reply:
ok, pls sync with QA for real case.